include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2039,6 +2039,71 @@ TriaAccessor<structdim, dim, spacedim>::diameter () const
 }
 
 
+template <int structdim, int dim, int spacedim>
+std::pair<Point<spacedim>,double>
+TriaAccessor<structdim, dim, spacedim>::smallest_enclosing_sphere () const
+{
+  switch (structdim)
+    {
+    case 1:
+      const Point<spacedim> p10 = this->vertex(1)-this->vertex(0);
+      return std::make_pair(this->vertex(0)+0.5*p10, p10.norm());
+    case 2:
+      const Point<spacedim> p30 = this->vertex(3)-this->vertex(0);
+      const Point<spacedim> p21 = this->vertex(2)-this->vertex(1);
+
+      const double diameter =0.;
+      const Point<spacedim> center;
+
+      (p30.norm() > p21.norm()) ?
```

Comment:
i think this can be written more concise and easier to read by
```
if (p30.norm() > p21.norm())
  return std::make_pair (this->vertex(0) + 0.5*p30, p30.norm())
else
  return std::make_pair (this->vertex(1) + 0.5*p21, p21.norm())
```